File: Leetcode/2364. Count Number of Bad Pairs.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        f={}
        c=0
        n=len(nums)
        for i in range(0,n):
            if (i-nums[i]) not in f:
                f[i-nums[i]]=1
            else:
                c+=f[i-nums[i]]
                f[i-nums[i]]+=1 
        return (n*(n-1))//2 - c
        

File: Leetcode/2364. Count Number of Bad Pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        long long n=nums.size();
        long long res=0;
        unordered_map<int,int> m;
        for(int i=0;i<n;i++){
            m[i-nums[i]]++;
            res+=(m[i-nums[i]]-1);
        }
        return (n*(n-1))/2 - res;
    }
};
